doc/compiler/il/GlRegDeps.md

```diff
@@ -0,0 +1,152 @@
+
+```
+n803n     ificmpne --> block_44 BBStart at n781n ()                                           [0x7f9d0f08eab0] bci=[-1,119,600] rc=0 vc=1295 vn=- li=-1 udi=- nc=3 flg=0x20
+n801n       iand                                                                              [0x7f9d0f08ea10] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=2
+n799n         iloadi  <isClassFlags>[#287  Shadow +36] [flags 0x603 0x0 ]                     [0x7f9d0f08e970] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n798n           aloadi  <componentClass>[#282  Shadow +88] [flags 0x10607 0x0 ]               [0x7f9d0f08e920] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n797n             aloadi  <vft-symbol>[#288  Shadow] [flags 0x18607 0x0 ]                     [0x7f9d0f08e8d0] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n106n               ==>aload
+n800n         iconst 1024                                                                     [0x7f9d0f08e9c0] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=0
+n802n       iconst 0                                                                          [0x7f9d0f08ea60] bci=[-1,119,600] rc=1 vc=1295 vn=- li=- udi=- nc=0
+n804n       GlRegDeps ()                                                                      [0x7f9d0f08eb00] bci=[-1,111,600] rc=1 vc=1295 vn=- li=- udi=- nc=3 flg=0x20
+n805n         PassThrough rcx                                                                 [0x7f9d0f08eb50] bci=[-1,117,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n111n           aload  tmp<auto slot 4>[#358  Auto] [flags 0x7 0x0 ]                          [0x7f9d0f081270] bci=[-1,117,600] rc=6 vc=1295 vn=- li=1 udi=70 nc=0
+n806n         PassThrough r8                                                                  [0x7f9d0f08eba0] bci=[-1,116,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n110n           iadd (cannotOverflow )                                                        [0x7f9d0f081220] bci=[-1,116,600] rc=5 vc=1295 vn=- li=1 udi=- nc=2 flg=0x1000
+n107n             ==>arraylength
+n109n             iconst -1 (X!=0 X<=0 )                                                      [0x7f9d0f0811d0] bci=[-1,115,600] rc=1 vc=1295 vn=- li=- udi=- nc=0 flg=0x204
+n807n         PassThrough rdi                                                                 [0x7f9d0f08ebf0] bci=[-1,111,600] rc=1 vc=1295 vn=- li=- udi=- nc=1
+n106n           ==>aload
+n796n     BBEnd </block_13>                                                                   [0x7f9d0f08e880] bci=[-1,119,600] rc=0 vc=1295 vn=- li=-1 udi=- nc=0
+```
+
+These nodes should be anchored under `xRegStore` earlier in the block. By anchoring these nodes earlier in the block, it ensures that they get evaluated earlier than `GlRegDeps` so that it does not interfere with the control flow. For example, `n106n` is now anchored under `aRegStore edi`. `n111n` is anchored under `aRegStore ecx`. `n110n` is anchored under `iRegStore r8d`.
```

Comment:
Should "should be anchored" be "must be anchored"?  That is, is this an absolute requirement, or are there exceptions?